### ISSUE TITLE
Compilation crashes on macOS 10.14.3 with GNU 8.3.0 due to errors in `strncpy`

### ISSUE THREAD

**Issue by mabalenk** [intel/mkl-dnn#432]
Dear all,

I'm attempting to compile MKL-DNN on a macOS platform using the GNU 8.3.0 compiler toolchain. Unfortunately, the compilation crashes due to the errors in `strncpy()` function, e.g.:

In function 'int mkldnn::impl::getenv(const char*, char*, int)',
    inlined from 'int mkldnn::impl::getenv_int(const char*, int)' at /Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:73:15:
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:57:20: error: 'char* strncpy(char*, const char*, size_t)' destination unchanged after copying no bytes [-Werror=stringop-truncation]
             strncpy(buffer, value, value_length);
             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

### Environment
* CPU make and model Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
* OS version Darwin galerkin.local 18.2.0 Darwin Kernel Version 18.2.0: Thu Dec 20 20:46:53 PST 2018; root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64 i386 MacBookPro11,4 Darwin
* Compiler version gcc (MacPorts gcc8 8.3.0_0) 8.3.0
* MKLROOT value /opt/intel/compilers_and_libraries_2018.2.164/mac/mkl
* CMake version cmake version 3.14.0
* git hash e35877d25f0d1d652ade669c2efa066c0d15c0d9

### Steps to reproduce
`CC=gcc CXX=g++ CMAKE_INSTALL_PREFIX=/opt/intel cmake -G "Unix Makefiles" ..`
`make`

### Actual behavior
The compilation crashes due to (a potentially) incorrect usage of `strncpy` function:

[ 10%] Building CXX object src/CMakeFiles/mkldnn.dir/common/utils.cpp.o
In function 'int mkldnn::impl::getenv(const char*, char*, int)',
    inlined from 'int mkldnn::impl::getenv_int(const char*, int)' at /Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:73:15:
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:57:20: error: 'char* strncpy(char*, const char*, size_t)' destination unchanged after copying no bytes [-Werror=stringop-truncation]
             strncpy(buffer, value, value_length);
             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:57:20: error: 'char* strncpy(char*, const char*, size_t)' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp: In function 'int mkldnn::impl::getenv_int(const char*, int)':
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:44:46: note: length computed here
     value_length = value == NULL ? 0 : strlen(value);
                                        ~~~~~~^~~~~~~
In function 'int mkldnn::impl::getenv(const char*, char*, int)',
    inlined from 'int mkldnn::impl::getenv_int(const char*, int)' at /Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:73:15,
    inlined from 'bool mkldnn::impl::jit_dump_enabled()' at /Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:121:35:
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:57:20: error: 'char* strncpy(char*, const char*, size_t)' destination unchanged after copying no bytes [-Werror=stringop-truncation]
             strncpy(buffer, value, value_length);
             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:57:20: error: 'char* strncpy(char*, const char*, size_t)' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp: In function 'bool mkldnn::impl::jit_dump_enabled()':
/Users/mabalenk/repo/git/mkl-dnn/src/common/utils.cpp:44:46: note: length computed here
     value_length = value == NULL ? 0 : strlen(value);
                                        ~~~~~~^~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/mkldnn.dir/build.make:427: src/CMakeFiles/mkldnn.dir/common/utils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:123: src/CMakeFiles/mkldnn.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

### Expected behavior
Flawless compilation.

**Comment by rsdubtso** [comment]
Should not actually be a bug because the string is zero-terminated at a later point. I will fix this as soon as I get a hold of 8.3.0 to test this with.

**Comment by rsdubtso** [comment]
Reproduced.
